network/src/commonMain/kotlin/com/wire/kalium/network/api/user/client/RegisterClientRequest.kt

```diff
@@ -7,53 +7,95 @@ import kotlinx.serialization.Serializable
 @Serializable
 data class RegisterClientRequest(
     @SerialName("password") val password: String,
+    @SerialName("prekeys") val preKeys: List<PreKey>,
+    @SerialName("lastkey") val lastKey: PreKey,
     @SerialName("class") val deviceType: DeviceType,
     @SerialName("type") val type: ClientType, // 'temporary', 'permanent', 'legalhold'
     @SerialName("label") val label: String,
-    @SerialName("prekeys") val preKeys: List<PreKey>,
-    @SerialName("lastkey") val lastKey: PreKey
+    @SerialName("capabilities") val capabilities: List<ClientCapability>?,
+    @SerialName("model") val model: String
 )
 
 @Serializable
 data class RegisterClientResponse(
     @SerialName("id") val clientId: String,
     @SerialName("type") val type: ClientType,
-    @SerialName("time") val registrationTime: String,
-    @SerialName("location") val location: LocationResponse? = null,
+    @SerialName("time") val registrationTime: String, // yyyy-mm-ddThh:MM:ss.qqq
+    @SerialName("location") val location: LocationResponse?,
     @SerialName("class") val deviceType: DeviceType?,
     @SerialName("label") val label: String?,
-    val capabilities: Capabilities?
+    @SerialName("cookie") val cookie: String?,
+    @SerialName("capabilities") val capabilities: Capabilities?,
+    @SerialName("model") val model: String?
 )
 
 @Serializable
 data class Capabilities(
-    @SerialName("capabilities") val capabilities: List<String>
+    @SerialName("capabilities") val capabilities: List<ClientCapability>
 )
 
 @Serializable
 enum class ClientType {
     @SerialName("temporary")
-    Temporary,
+    Temporary {
+        override fun toString(): String {
+            return "temporary"
+        }
+    },
 
     @SerialName("permanent")
-    Permanent,
+    Permanent {
+        override fun toString(): String {
+            return "permanent"
+        }
+    },
 
     @SerialName("legalhold")
-    LegalHold
+    LegalHold {
+        override fun toString(): String {
+            return "legalhold"
+        }
+    }
 }
 
 @Serializable
 enum class DeviceType {
     //'phone', 'tablet', 'desktop', 'legalhold'
     @SerialName("phone")
-    Phone,
+    Phone {
+        override fun toString(): String {
+            return "phone"
+        }
+    },
 
     @SerialName("tablet")
-    Tablet,
+    Tablet {
+        override fun toString(): String {
+            return "tablet"
+        }
+    },
 
     @SerialName("desktop")
-    Desktop,
+    Desktop {
+        override fun toString(): String {
+            return "desktop"
+        }
+    },
 
     @SerialName("legalhold")
-    LegalHold
+    LegalHold {
+        override fun toString(): String {
+            return "legalhold"
+        }
+    }
```

Comment:
What is the `toString` necessary for?

Maybe can we just override the enum class `toString`?

```kotlin
enum class Foo {
BAR, FOOBAR;

    override fun toString(): String {
        return this.name.lowercase()
    }
}
```

Reply:
it's for testing purposes , i guess override the enum class toString will work in most cases
i will change it